packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -205,19 +205,22 @@ export function FileUploader({
       prevFileStatuses.map((status, index) => ({
         ...status,
         uploadTask: uploadTasksTemp?.[index],
-        fileState: status.fileState ?? 'loading',
+        fileState:
+          status.fileState === FileState.INIT
+            ? FileState.LOADING
+            : status.fileState,
```

Comment:
Before it was checking if the file was undefined or null. Now it only changes the file state if it's in the `INIT` state. That way it doesn't change the file state to loading for files that are in any other state.